pkg/controller/uniteddeployment/adapter/cloneset_adapter.go

```diff
@@ -135,6 +139,35 @@ func (a *CloneSetAdapter) ApplySubsetTemplate(ud *alpha1.UnitedDeployment, subse
 	attachNodeAffinity(&set.Spec.Template.Spec, subSetConfig)
 	attachTolerations(&set.Spec.Template.Spec, subSetConfig)
 
+	if len(subSetConfig.Patch.Raw) != 0 {
+		cloneBytes, _ := json.Marshal(set)
+		modified, err := strategicpatch.StrategicMergePatch(cloneBytes, subSetConfig.Patch.Raw, &alpha1.StatefulSet{})
```

Comment:
Have you test it in a real cluster? (or you can write an e2e case in `test/e2e/apps`) I'm not sure whether you can use `strategicpatch.StrategicMergePatch` for cloneset and advanced statefulset, since they are defined by CRD.

Reply:
I have tested and I add test in `pkg/controller/uniteddeployment`. Do you mean need to  add `uniteddeployment.go `in `test/e2e/apps `?

Reply:
Yes, add an e2e case so we can test it in a real k8s cluster.

Reply:
@wxyh Hi boy, can you write an e2e case in "test/e2e/apps" as FillZpp said? I really like the feature you submitted, and I want to use it in the upcoming new version of Kruise, otherwise I would need to resubmit the pr myself.

Reply:
@shijieqin OK. I'll add it in the next few days

Reply:
@FillZpp I found some problems in the test.
1.  For `cloneSet` and `advancedStatefulSet`, some fields such as `ScaleStrategy`,`MinReadySeconds`,`Lifecycle`  don't assign value from `CloneSetTemplate` and `AdvancedStatefulSetTemplate` .Is this a mistake? should I fix it in another pr?
2. `strategicpatch.StrategicMergePatch` can be used in `cloneSet` and `advancedStatefulSet`.But for list like `VolumeClaimTemplates`  patch strategy is `replace` because `patchStrategy` is not defined. I think that's acceptable,what do you think ?